### DISCUSSION TITLE
How to better cooperate with Vuex and Vue Router in <script setup></script> in 2.7 #12651

### DISCUSSION THREAD

**Discussion by 2392645016** [2392645016/vue#10] [Q&A] [answered]
What problem does this feature solve?
Hello！How to better cooperate with Vuex and Vue Router in <script setup></script> in 2.7？

**Comment by 202780181** [comment, marked as answer]
Vuex: Migrate to [pinia](https://github.com/vuejs/pinia)
Router: https://github.com/vuejs/vue-router/issues/3760

**Reply by 202780181** [reply]
[yyx990803](https://github.com/yyx990803)

**Comment by 2392645016** [comment]
Thanks for your advice.
